react-navigation/react-navigation
# fix: using initialRouteName in web, make screen redirects to pages outside app

Please provide enough information so that others can review your pull request:

**Motivation**
If we set initialRouteName and directly access the deep link, back button on the initial screen keeps redirecting us to page outside app.
Preview

https://user-images.githubusercontent.com/24470609/156485734-d4de463c-6562-4317-bbad-bd9d217d7b02.mp4

This PR added a guard to avoid going back outside the app when using App Back Button & supplying initialRouteName. Should address this issue https://github.com/react-navigation/react-navigation/issues/10398

**Test plan**

- Create patch for this fix
- Apply patch to the reproducible bug (I've applied it in here - https://github.com/listiani13/react-navigation-initial-route-name-bug/tree/poc-using-fix)

Preview:

https://user-images.githubusercontent.com/24470609/157353361-3cab3942-a861-48cd-8bde-a9114b466f87.mp4


> Make sure you test on both platforms if your change affects both platforms.

It's not affecting native, because I only change `useLinking.ts` (not `.native.tsx`) 

> The code must pass tests.

✅

**Code formatting**

> Look around. Match the style of the rest of the codebase. Run `yarn lint --fix` before committing.

✅
